src/server/pkg/hashtree/hashtree.go

```diff
@@ -530,17 +534,24 @@ func (h *hashtree) putFile(path string, objects []*pfs.Object, overwriteIndex *p
 	if overwriteIndex != nil && overwriteIndex.Index <= int64(len(node.FileNode.Objects)) {
 		node.FileNode.Objects = node.FileNode.Objects[:overwriteIndex.Index]
 	}
-	node.SubtreeSize += sizeDelta
+	node.SubtreeSize += sizeDelta + headerFooterSize
 	node.FileNode.Objects = append(node.FileNode.Objects, objects...)
 	h.changed[path] = true
 
 	// Add 'path' to parent (if it's new) & mark nodes as 'changed' back to root
+	updatedParent := false
 	return h.visit(path, func(node *NodeProto, parent, child string) error {
 		if node == nil {
 			node = &NodeProto{
 				Name:    base(parent),
 				DirNode: &DirectoryNodeProto{},
 			}
+			if !updatedParent {
```

Comment:
This would be a bit clearer, and a bit shorter if you just check to see if `parent == filepath.Dir(path)` that's the intent here right? You only want to attach the header / footer to the direct parent of the path being put.

Reply:
Makes sense. Fixed.